sdk/dotnet/Storage/SharedAccessSignature.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Copyright 2016-2018, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+
+namespace Pulumi.Azure.Storage
+{
+    /// <summary>
+    /// Utility functions to manipulate Azure Storage Shared Access Signatures (SAS).
+    /// </summary>
+    public static class SharedAccessSignature
+    {
+        /// <summary>
+        /// Produce a URL with read-only access to a Storage Blob with a Shared Access Signature (SAS).
+        /// </summary>
+        /// <param name="blob">Blob to construct the URL for.</param>
+        /// <param name="account">Storage account containing the blob.</param>
+        /// <returns>An <see cref="Output{String}"/> containing the blob access URL.</returns>
+        public static Output<string> SignedBlobReadUrl(ZipBlob blob, Account account)
+        {
+            return Output
+                .All<string>(account.Name, account.PrimaryConnectionString, blob.StorageContainerName, blob.Name)
+                .Apply(async values =>
+                {
+                    string accountName = values[0];
+                    string connectionString = values[1];
+                    string containerName = values[2];
+                    string blobName = values[3];
```

Comment:
this is totally fine.  another option would be to do `.Tuple` and then `async ((string accountName, string connecitonString, etc.) t`.  Up to you though.

Reply:
I suspect there was no 4-way tuple when I did that. I'll adjust.